### PR TITLE
docs(web-modeler): introduce gitlab sync

### PR DESCRIPTION
**Pull request by marcellobarile**
## Description

Introduces the GitLab sync functionality, which was added to the existing Git Sync feature.

Epic: https://github.com/camunda/product-hub/issues/2469
Part of https://github.com/camunda/web-modeler/issues/10911

## When should this change go live?

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**.
- [ ] This is already available but undocumented and should be released within a week.
- [x] This on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (apply `hold` label or convert to draft PR)
- [x] This is part of a scheduled alpha or minor. (apply alpha or minor label)
- [ ] There is **no urgency** with this change and can be released at any time.

## PR Checklist

- [ ] My changes are for an **already released minor** and are in `/versioned_docs` directory.
- [x] My changes are for the **next minor** and are in `/docs` directory (aka `/next/`).

- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned the Engineering DRI or delegate.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @camunda/tech-writers as a reviewer.

### CODE REVIEW

**Review by conceptualshark — DISMISSED** [review]
I've left a few suggestions here in two separate commits:

- A review of the content as-is [here](https://github.com/camunda/camunda-docs/pull/4583/commits/f6e34eecc9da909759e4097c56b472ac324d3a43)
- An option condensing the information into a tabbed display [here](https://github.com/camunda/camunda-docs/pull/4583/commits/dd57d011bbc1408d285f1ac53ffe719315360d56)

If you'd prefer not to introduce tabs at this time, I can roll back the second commit. You should be able to see the change in the deployed preview site, like this:
Screenshot 2024-11-08 at 12 29 47 PM

If any other repository hosts are planned, it will be helpful to have this scaffolding in place to add more options. Alternatively, we could look at creating a generic "Git Sync" page, and add options as their own independent subpages and reusable content. I'd be happy to take on some of this content structuring if it ends up being necessary, and it doesn't need to be decided or implemented here.

Otherwise this looks good to me! I'm approving for now, but can edit/adjust if the structure change isn't warranted at this time.

**Comment by marcellobarile** [comment]
@conceptualshark 
> An option condensing the information into a tabbed display

Yes! That's fantastic! We will indeed add more providers in the (near) future. Thanks!

File: docs/components/modeler/web-modeler/git-sync.md [hunk 1]
```diff
@@ -65,15 +80,67 @@ When successful, your project will display a new **Sync with GitHub** button.
...
+Web Modeler requires a project access token to sync changes with your GitLab repository.
+
+Follow the [GitLab documentation](https://docs.gitlab.com/ee/user/project/settings/project_access_tokens.html#create-a-project-access-token) to generate a new project access token for your repository with the following configuration:
+
+- Enable the following [**scopes**](https://docs.gitlab.com/ee/user/project/settings/project_access_tokens.html#scopes-for-a-project-access-token):
+  - `api`
+  - `read_repository`
+  - `write_repository`
+  - `read_registry`
+  - `write_registry`
```
Comment (JacobiCamunda) [inline docs/components/modeler/web-modeler/git-sync.md:97]: 🔧 Some of these permission are not needed (in my opinion). `api`, `read_api`, `read_repository`, `write_repository` should be enough but correct my if I missed something.

**Review by JacobiCamunda — DISMISSED** [review]
Looks good from my side, except one thing about the permissions.
But already approving 👍

File: docs/components/modeler/web-modeler/git-sync.md (hunk 1 above)
Comment (JacobiCamunda) [reply docs/components/modeler/web-modeler/git-sync.md:97]: This is what I used for testing. When the implementation is done I will revisit the page and update if we need more permissions than I assumed.

Comment (marcellobarile) [reply docs/components/modeler/web-modeler/git-sync.md]: okay, they are udpated now.

**Review by conceptualshark — DISMISSED** [review]

**Review by conceptualshark — APPROVED** [review]
👍 This update looks good to me!

**Comment by christinaausley** [comment]
@marcellobarile Feel free to mark this as `Ready fro review` and merge when you are ready 👍

**Pull request merged**